src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml

```diff
@@ -363,7 +363,6 @@ well as menu structures (for main menu and popup menus).
          <menu label="_Import Dataset">
             <cmd refid="importDatasetFromFile"/>
             <cmd refid="importDatasetFromURL"/>
-            <cmd refid="importDatasetFromCSV"/>
```

Comment:
This should be quite straightforward. The feature flag should be checked during Application.initializeWorkbench and the unsupported commands removed (via Command.remove()). This will automatically remove them from all menus. So in short, you should add all data import commands to both the main and toolbar menus (unqualified) and then do the dynamic removal of commands within Application.initializeWorkbench.